### PR TITLE
add api changes according to gep-3155

### PR DESCRIPTION
**Pull request by LiorLieberman**
/kind feature
**What this PR does / why we need it**:
Update CRDs per [gep-3155](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-3155/index.md)

**Does this PR introduce a user-facing change?**:

```release-note
Enhance experimental support for backendmtls configuration
```

/cc @mkosieradzki

### CODE REVIEW

File: apis/v1/shared_types.go
Comment (LiorLieberman) [inline apis/v1/shared_types.go:549]: Not sure how we should approach for case-sensitivity, scheme and host aren't but other parts _can_ be. Let me know if you want to add any language to indicate what we expect

File: apis/v1/shared_types.go
Comment (LiorLieberman) [inline apis/v1/shared_types.go:544]: Not sure how we should approach for case-sensitivity, scheme and host aren't but other parts can be. Let me know if you want to add any language to indicate what we expect

Comment (youngnick) [reply apis/v1/shared_types.go:544]: Maybe "implementations SHOULD treat this string case insensitively" or something? Feels like it's better to have _some_ stance on case sensitivity rather than none.

File: pkg/test/cel/grpcroute_experimental_test.go
Comment (youngnick) [inline pkg/test/cel/grpcroute_experimental_test.go:1]: Seems like only whitespace changes in this file? Was this intended or is this an editor thing?

File: pkg/test/cel/httproute_experimental_test.go
Comment (youngnick) [inline pkg/test/cel/httproute_experimental_test.go:1]: More whitespace-only changes?

**Review by youngnick — COMMENTED** [review]
Changes LGTM, aside from extra whitespace.

File: apis/v1/gateway_types.go
Comment (mlavacca) [inline apis/v1/gateway_types.go:239]: I was kind of confused when I saw this new experimental feature to be set as core. This can be problematic, as when we graduate this feature to GA, this will be a breaking change: all implementation will have to support it, otherwise conformance won't be given. I think this is a bit problematic, but I'd like to ask others' opinion (@robscott @youngnick @shaneutt).

I just found out we already have a precedent, with the [infrastructure Gateway feature](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/gateway_types.go#L225-L231).

Comment (LiorLieberman) [reply apis/v1/gateway_types.go:239]: Fair. I think before graduating this to GA we should have at least (3?) implementations implementing it, right? 

I am not sure if there are guidelines on whether a feature should have to graduate as extended, I think support level and channel are two different things. I believe if we move _any_ feature to Core support, it would mean some implementations would loose conformance.

File: pkg/test/cel/grpcroute_experimental_test.go
Comment (LiorLieberman) [reply pkg/test/cel/grpcroute_experimental_test.go:1]: 👍

File: apis/v1/shared_types.go
Comment (robscott) [reply apis/v1/shared_types.go:544]: I think I'd leave case sensitivity out of this as it's potentially different depending on the scheme. Per RFC-3986:

>When a URI uses components of the generic syntax, the component syntax equivalence rules always apply; namely, that the scheme and host are case-insensitive and therefore should be normalized to lowercase.  For example, the URI <HTTP://www.EXAMPLE.com/> is equivalent to <http://www.example.com/>.  The other generic syntax components are assumed to be case-sensitive unless specifically defined otherwise by the scheme

https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2.1

File: apis/v1/shared_types.go
Comment (robscott) [inline apis/v1/shared_types.go:548]: Non-blocking nit: Would be nice to have a way to track the source of this RegEx for future readers. Can you leave a comment in https://github.com/kubernetes-sigs/gateway-api/issues/3306 with the source of this validation?

File: apis/v1/gateway_types.go
Comment (shaneutt) [reply apis/v1/gateway_types.go:239]: I'm in favor of starting in extended and then we can start communicating a desire to move it to core as soon as we like, giving some soak time for implementations to think about that.

Comment (robscott) [reply apis/v1/gateway_types.go:239]: My thinking is that nothing in experimental channel can really be "core" anyways, so the support level here is more aspirational than anything. So at this point we're signaling that we think this can be "core", and thus giving the maximum possible notice that that's what it may graduate to standard as. Any alternative seems like it would either result in less notice to implementations or graduating the feature as "extended" and then having to manage a second transition to "core". I'd rather just start with a target of what we think we can achieve and then accept pushback to a lower level before this feature graduates if we think that's necessary.

**Review by robscott — COMMENTED** [review]
Thanks @LiorLieberman! Will defer to someone else for final LGTM.

/approve

File: apis/v1/gateway_types.go
Comment (shaneutt) [reply apis/v1/gateway_types.go:239]: +1

**Review by shaneutt — APPROVED** [review]
Thanks @LiorLieberman!

This looks good:

/approve

However, there seems to be some building consensus that we should not be calling this supported at a `Core` level so just putting a hold until everyone's had a chance to say their piece and we sort that out.

/hold

**Comment by LiorLieberman** [comment]
> Thanks @LiorLieberman!
> 
> 
> 
> This looks good:
> 
> 
> 
> /approve
> 
> 
> 
> However, there seems to be some building consensus that we should not be calling this supported at a `Core` level so just putting a hold until everyone's had a chance to say their piece and we sort that out.
> 
> 
> 
> /hold

I saw you commented +1 on the thread to leave it as core, to indicate our aspirational support level. Am i wrong ?

**Comment by robscott** [comment]
> I saw you commented +1 on the thread to leave it as core, to indicate our aspirational support level. Am i wrong ?

Maybe just waiting to ensure @mlavacca is on board with this approach? In any case, I'm fine to hold off on merging this until he can take a look. IMO, this is sufficiently complete within the timeline we'd agreed to to make it into v1.2, just holding off on final merge for one last signoff.

File: apis/v1/gateway_types.go
Comment (mlavacca) [reply apis/v1/gateway_types.go:239]: I'm fine with having this targeted as core, and agree when you say that in case we'll change it in the future, the core->extended transition is less problematic than the extended->core. One thing we should be very careful about is clearly stating to implementations that when an experimental core feature graduates, that's a breaking change (i.e., all the implementations MUST implement it). We need to communicate this ahead of time, to avoid that implementation suddenly and inexpectedly see their conformance broken.

**Review by mlavacca — COMMENTED** [review]
I'm fine with merging this one. I left a comment about the core vs extended discussion in the related thread.

/lgtm
/unhold

**Pull request merged**
